src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -2485,12 +2491,43 @@ export class StudyViewPageStore {
                             .clinicalAttributeId,
                     } as ClinicalDataCountItem);
                     let counts: ClinicalDataCount[] = [];
+                    let attributeId: string = '';
                     if (data !== undefined) {
                         counts = data.counts;
+                        attributeId = data.attributeId;
+                        if (!this.chartToUsedColors.has(attributeId))
+                            this.chartToUsedColors.set(attributeId, new Set());
                     }
-                    return getClinicalDataCountWithColorByClinicalDataCount(
+
+                    let res = getClinicalDataCountWithColorByClinicalDataCount(
                         counts
                     );
+                    res.map((item: any) => {
```

Comment:
while the effect of `map` as you're using it will be the same as `forEach`, it's more semantic to use `forEach` since it more easily indicates that the purpose of the loop is to iterate rather than to return a new array